composer/algorithms/factorize/factorize_modules.py

```diff
@@ -0,0 +1,441 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+import abc
+import math
+from typing import Optional, Tuple, Union, cast
+
+import numpy as np
+import torch
+from torch import nn
+from torch.nn.common_types import _size_2_t
+
+from composer.algorithms.factorize.factorize_core import LowRankSolution, factorize_conv2d, factorize_matrix
+
+FractionOrInt = Union[int, float]
```

Comment:
Would it be possible to hardcode `Union[int, float]` where it is used? I get this simplifies things code-wise, but makes it slightly less inteutivie to follow it, as I had to look up in another file what the allowed types actually are (by `Fraction`, I was originally thinking a https://docs.python.org/3/library/fractions.html)